samples/comeasy/wrotei.cpp

```diff
@@ -101,7 +101,7 @@ int WINAPI TimedEntryPoint(VOID)
     }
     else {
         printf("wrotei" DETOURS_STRINGIFY(DETOURS_BITS) ".dll:"
-               " Error detouring IStram::Wrote(): %d\n", error);
+               " Error detouring IStram::Wrote(): %ld\n", error);
```

Comment:
I've long been ambivalent about this sort of mismatch, because:
 a. They have the same underlying size.
 b. Signedness is still mixed up, but perhaps deliberately to affect the output format.
 c. Maybe some other system typedefs `DWORD` to be unsigned int instead, still 32bits, and the code is portable to that.
So the whole printf thing is broken really. Thinking more broadly, the format string is overly tied to the concrete type of input. Which could be a typedef and there could be a desired flexiblity.

It makes you wonder if iostreams should be used, so the code can become type-ignorant.

Or a macro to abstract, like PRINTF_FORMAT_DWORD, like is provided by stdint.h/inttypes.h/etc. for many types.